src/components/molecules/Chatbox/Chatbox.tsx

```diff
@@ -1,106 +1,84 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaperPlane } from "@fortawesome/free-solid-svg-icons";
 
-import { WithId } from "utils/id";
-import { chatUsersSelector } from "utils/selectors";
+import { MessageToDisplay } from "types/chat";
+import { OnAvatarClick } from "types/User";
 
-import { useVenueId } from "hooks/useVenueId";
-import { useWorldUsersById } from "hooks/users";
-import { useSelector } from "hooks/useSelector";
-
-import { PrivateChatMessage, RestrictedChatMessage } from "store/actions/Chat";
-import ChatList from "../ChatList";
+import { ChatMessage } from "components/atoms/ChatMessage";
 
 import "./Chatbox.scss";
 
-interface ChatOutDataType {
-  messageToTheBand: string;
-}
-
-interface ChatboxProps {
-  chats: WithId<PrivateChatMessage | RestrictedChatMessage>[];
-  onMessageSubmit: (data: ChatOutDataType) => void;
-  allowDelete?: boolean;
-  emptyListMessage?: string;
-  showSenderImage?: boolean;
-  isVenueChat?: boolean;
-}
+export type ChatboxProps = {
+  messages: MessageToDisplay[];
+  sendMessage: (text: string) => void;
+  deleteMessage: (message: {}) => void;
+  onAvatarClick: OnAvatarClick;
+};
 
-// @debt TODO: we have a ChatBox in organisms but also in molecules.. are they the same? Can we de-dupe them?
-const ChatBox: React.FC<ChatboxProps> = ({
-  allowDelete,
-  chats,
-  onMessageSubmit,
-  emptyListMessage,
-  showSenderImage,
-  isVenueChat,
+export const Chatbox: React.FC<ChatboxProps> = ({
+  messages,
+  sendMessage,
+  onAvatarClick,
 }) => {
-  const venueId = useVenueId();
-  const [isMessageToTheBarSent, setIsMessageToTheBarSent] = useState(false);
-
-  const { worldUsersById } = useWorldUsersById();
-  const chatUsersById = useSelector(chatUsersSelector) ?? {};
-
-  const usersById = isVenueChat ? worldUsersById : chatUsersById;
+  const [isMessageBeingSent, setIsMessageBeingSent] = useState(false);
 
   useEffect(() => {
-    if (isMessageToTheBarSent) {
+    if (isMessageBeingSent) {
       setTimeout(() => {
-        setIsMessageToTheBarSent(false);
+        setIsMessageBeingSent(false);
       }, 2000);
     }
-  }, [isMessageToTheBarSent]);
+  }, [isMessageBeingSent]);
 
-  const { register, handleSubmit, reset } = useForm<ChatOutDataType>({
+  const { register, handleSubmit, reset } = useForm<{ message: string }>({
     mode: "onSubmit",
   });
 
-  const submitChatMessage = useCallback(
-    async (data: ChatOutDataType) => {
-      setIsMessageToTheBarSent(true);
-      onMessageSubmit(data);
-      reset();
-    },
-    [onMessageSubmit, reset]
-  );
+  const onSubmit = handleSubmit(({ message }) => {
+    setIsMessageBeingSent(true);
+    sendMessage(message);
+    reset();
+  });
 
-  const deleteMessage = useCallback(
-    async (id: string) => {
-      await firebase
-        .firestore()
-        .doc(`venues/${venueId}/chats/${id}`)
-        .update({ deleted: true });
-    },
-    [venueId]
+  const rendreredMessages = useMemo(
```

Comment:
I didn't dig into the logic of `ChatList` completely. I think it could be replaced with a component, but do you feel strongly about it? Chatbox component isn't large, so `rendered` approach doesn't clutch it too much.

Reply:
Worth taking a look through it, as you're more familiar with the chat code than me right now. But basically it seems to be handling all the stuff related to showing a list of chats + delete logic, etc.

My thoughts when I skimmed through this diff/changes here was it looked as though you might be re-implementing the same functionality.

Reply:
I did skim through the old components(and reused the logic from there) and they had all the functionality that we have now, but all of that was in a terrible state, not reusable and hard to extend. With the codebase like that, it would be hard to add `World chat` in the future for example.

Reply:
I'm pretty sure `ChatList` was one of the ones I had already refactored/cleaned up, which is why i'm asking. Looking at it right now, it doesn't look like there is a lot of 'excess' stuff in it beyond what would be needed/useful here.

Can you please take another look and compare/contrast to see if it would be useful here, or if not, why it doesn't make sense to reuse it.

Reply:
It's 10 lines of code. The boilerplate for creating a new component would probably double or triple that
![image](https://user-images.githubusercontent.com/63194656/108505455-1e259980-72c0-11eb-9777-2d6e689c4645.png)

Reply:
The reason I was saying to use ChatList is that it also already handled all of the deletion/etc things within it. So it would be less 'boilerplate of adding code' and more 'use the thing that already exists and handles it', so likely would have been less code.

It's done now though, so _shrugs_